server/item/writable_book.go

```diff
@@ -0,0 +1,100 @@
+package item
+
+type WritableBook struct {
+	Pages []string
+}
+
+// Page returns a specific page from the book. If the page exists, it will return the content and true, otherwise
+// it will return an empty string and false.
+func (w WritableBook) Page(page uint) (string, bool) {
+	p := int(page)
+	if len(w.Pages) <= p {
+		return "", false
+	}
+	return w.Pages[p], true
+}
+
+// MaxCount always returns 1.
+func (w WritableBook) MaxCount() int {
+	return 1
+}
+
+// Exists checks to see weather a page exists or not
+func (w WritableBook) Exists(page uint) bool {
```

Comment:
Wrong spelling of whether, a name like PageExists() would also be more understandable. Same thing about the uint parameter